src/middleware/inject_params.rs

```diff
@@ -55,11 +55,15 @@ impl Middleware<CallRequest, Result<JsonValue, Error>> for InjectParamsMiddlewar
                 // full params with current block
                 return next(request).await;
             }
-            len if len == idx => {
+            len if len <= idx => {
                 // without current block
                 let to_inject = self.get_parameter().await;
                 tracing::debug!("Injected param {} to method {}", &to_inject, request.method);
+                while request.params.len() < idx {
+                    request.params.push(JsonValue::Null);
```

Comment:
we need to inject null if param is optional otherwise throw error invalid params

Reply:
it should be fine, the optional params are the last ones as idiom

Reply:
I think what @ermalkaleci say is we shouldn't inject if it is not optional and should just return invalid parameter error instead forward it to upstream